service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/configurations/serializers/ConfigurationContentParser.java

```diff
@@ -36,7 +33,7 @@
     @Setter
     private Map<String, Object> deviceContent;
 
-    private final transient static Gson gson = new Gson();
+    private final transient static Gson gson = new GsonBuilder().setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE).create();
```

Comment:
Adding objectToNumber strategy here so that ints are kept as ints instead of read/written as floats